include/rocksdb/file_system.h

```diff
@@ -733,6 +747,19 @@ struct FSReadRequest {
   IOStatus status;
 };
 
+// A read IO response structure for use by asynchronously Read APIs.
+struct FSReadResponse {
```

Comment:
Does this also need the "scratch" to store the actual data the Slice refers to?

Reply:
This might also need some sort of ID to identify the request that completed.

Reply:
The callback function will take void* which will be useful for identifying the request associated with the response.